pkg/aws/endpoints/resolver.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoints
+
+import (
+	operatorOption "github.com/cilium/cilium/operator/option"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/endpoints"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "aws-endpoints")
+
+	defaultResolver = endpoints.NewDefaultResolver()
+)
+
+func Resolver(service, region string) (aws.Endpoint, error) {
+	if ep := operatorOption.Config.EC2APIEndpoint; len(ep) > 0 && service == "ec2" {
+		log.Debugf("Using custom API endpoint %s for service %s in region %s", ep, service, region)
+		// See https://docs.aws.amazon.com/sdk-for-go/v2/api/aws/endpoints/#hdr-Using_Custom_Endpoints
+		return aws.Endpoint{
+			URL: "https://" + ep,
```

Comment:
Do we need to prepend the `https://` prefix here? I see that in the docs you linked we can provide URL without any scheme, like:

```
aws.Endpoint{
        URL:           "s3.custom.endpoint.com",
        SigningRegion: "custom-signing-region",
}
```

I think it might be good to use the endpoint provided by the user as it is without adding a scheme as we don't do any other validation here. So let's say if a user provides a URL with `https://` already prefixed this will not work. If we don't add a scheme user can provide URL either without the scheme or with the scheme.

Reply:
The linked docs are a bit misleading in that respect. In my tests, the endpoint URL was only working if a scheme was provided. Given that `https://` is the only one that worked here, I thought it's easier to just hard-code this and let the user only specify the hostname.

Reply:
In that case, it makes sense, for now, to go with this. This shouldn't be a problem as in the example of the command-line flag we do provide the right kind of endpoint we are expecting. We can always circle back to this later if anyone faces the issue.

Reply:
Sounds good, thank you for the review @fristonio!